components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -2826,7 +2828,14 @@ private static void clearTokenCacheEntry(String consumerKey, Set<AccessTokenDO>
             throws IdentityOAuthAdminException {
 
         for (AccessTokenDO detailToken : activeDetailedTokens) {
-            clearTokensFromCache(consumerKey, detailToken, detailToken.getAccessToken());
+            String accessToken = detailToken.getAccessToken();
+            try {
+                accessToken = OAuth2Util.getPersistenceProcessor().getPreprocessedAccessTokenIdentifier(accessToken);
```

Comment:
if EncryptionDecryptionPersistenceProcessor is not enabled, I hope this method will return unencrypted access token